packages/marketplace/contracts/RoyaltiesRegistry.sol

```diff
@@ -110,23 +111,22 @@ contract RoyaltiesRegistry is OwnableUpgradeable, IRoyaltiesProvider {
     /// @notice returns royalties type from uint
     /// @param data in uint256
     /// @return royalty type
-    function _getRoyaltiesType(uint256 data) internal pure returns (uint256) {
-        for (uint256 i = 1; i <= ROYALTIES_TYPES_AMOUNT; ++i) {
+    function _getRoyaltiesType(uint256 data) internal pure returns (RoyaltyTypes) {
+        for (uint256 i = 1; i <= uint256(type(RoyaltyTypes).max); ++i) {
             if (data / 2 ** (256 - i) == 1) {
-                return i;
+                return RoyaltyTypes(i);
             }
         }
-        return ROYALTIES_TYPE_UNSET;
+        return RoyaltyTypes.UNSET;
     }
 
     /// @notice sets royalties type for token contract
     /// @param token address of token
     /// @param royaltiesType uint256 of royalty type
     /// @param royaltiesProvider address of royalty provider
-    function _setRoyaltiesType(address token, uint256 royaltiesType, address royaltiesProvider) internal {
-        require(royaltiesType > 0, "wrong royaltiesType");
-        require(royaltiesType <= ROYALTIES_TYPES_AMOUNT, "invalid royaltiesType amount");
-        royaltiesProviders[token] = uint256(uint160(royaltiesProvider)) + 2 ** (256 - royaltiesType);
+    function _setRoyaltiesType(address token, RoyaltyTypes royaltiesType, address royaltiesProvider) internal {
```

Comment:
naming not consistent, we did singular for FeeSide and AssetClassType, so probably keep enverything singular. Also , it's always using `royalties` not `royalty` in the contract:
RoyaltyTypes royaltiesType => RoyaltiesType royaltiesType ?